lnbits/wallets/phoenixd.py

```diff
@@ -0,0 +1,169 @@
+        password = settings.phoenixd_api_password
+        encoded_auth = base64.b64encode(f":{password}".encode())
+        auth = str(encoded_auth, "utf-8")
+        self.headers = {
+            "Authorization": f"Basic {auth}",
+            "User-Agent": settings.user_agent,
+        }
+
+        self.client = httpx.AsyncClient(
+            base_url=self.endpoint, auth=("", settings.phoenixd_api_password)
+        )
+
+    async def cleanup(self):
+        try:
+            await self.client.aclose()
+        except RuntimeError as e:
+            logger.warning(f"Error closing wallet connection: {e}")
+
+    async def status(self) -> StatusResponse:
+        try:
+            r = await self.client.get("/getinfo", timeout=10)
+        except (httpx.ConnectError, httpx.RequestError):
+            return StatusResponse(f"Unable to connect to '{self.endpoint}'", 0)
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return StatusResponse(error_message, 0)
+
+        data = int(r.json()["channels"][0]["balanceSat"]) * 1000
+        return StatusResponse(None, data)
+
+    async def create_invoice(
+        self,
+        amount: int,
+        memo: Optional[str] = None,
+        description_hash: Optional[bytes] = None,
+        unhashed_description: Optional[bytes] = None,
+        **kwargs,
+    ) -> InvoiceResponse:
+        if description_hash or unhashed_description:
+            raise Unsupported("description_hash")
+
+        msats_amount = amount
+        data: Dict = {
+            "amountSat": f"{msats_amount}",
+            "description": memo,
+            "externalId": "",
+        }
+
+        r = await self.client.post(
+            "/createinvoice",
+            data=data,
+            timeout=40,
+        )
+
+        if r.is_error:
+            error_message = r.json()["message"]
+            return InvoiceResponse(False, None, None, error_message)
+
+        data = r.json()
+        # logger.info(f'data: {data}')
+
+        checking_id = data["paymentHash"]
+        payment_request = data["serialized"]
+        return InvoiceResponse(True, checking_id, payment_request, None)
+
+    async def pay_invoice(
+        self, bolt11_invoice: str, fee_limit_msat: int
+    ) -> PaymentResponse:
+        r = await self.client.post(
+            "/payinvoice",
+            data={
+                "invoice": bolt11_invoice,
+            },
+            timeout=40,
+        )
+
+        if r.is_error:
+            logger.error(f"pay_invoice error: {r.json()}")
+            error_message = r.json()["message"]
+            return PaymentResponse(False, None, None, None, error_message)
+
+        data = r.json()
+        logger.info(f"pay_invoice data: {data}")
+
+        checking_id = data["paymentHash"]
+        fee_msat = -int(data["routingFeeSat"])
+        preimage = data["paymentPreimage"]
+
+        return PaymentResponse(True, checking_id, fee_msat, preimage, None)
```

Comment:
This is very dangerous. `True` should never be returned directly but only through a dictionary that explicitly checks for a valid response from the backend (see other wallet files).

Reply:
This `True` is returned directly in https://github.com/lnbits/lnbits/blob/4f118c5f98247dce8509089a8c3660099df2bff3/lnbits/wallets/lntips.py#L122 and https://github.com/lnbits/lnbits/blob/4f118c5f98247dce8509089a8c3660099df2bff3/lnbits/wallets/opennode.py#L115 and https://github.com/lnbits/lnbits/blob/4f118c5f98247dce8509089a8c3660099df2bff3/lnbits/wallets/lnpay.py#L127 and https://github.com/lnbits/lnbits/blob/4f118c5f98247dce8509089a8c3660099df2bff3/lnbits/wallets/zbd.py#L120 and possibly a few other backend wallet implementations.  🧐

Please specify which wallet backend should be the correct model, there's at least 4 which are currently setup

@prusnak @motorina0

Reply:
@motorina0 @dni 

bump. what should i do?

Reply:
> This is very dangerous. `True` should never be returned directly but only through a dictionary that explicitly checks for a valid response from the backend (see other wallet files).

The [endpoint](https://phoenix.acinq.co/server/api#pay-invoice) does not return a `"success"` kind of flag (unfortunately):

**Response**
```json
{
    "recipientAmountSat": 1,
    "routingFeeSat": 4,
    "paymentId": "961d413c-91ca-4f3d-9d91-760a08405901",
    "paymentHash": "9586b2819f3c6d3c3e087984b97dcf4b74babfcaa5da4bc4207f47c6c1cc71e0",
    "paymentPreimage": "6a3e65d9a05113864c9c27e430cc99751359ccbfa8fd5869aea46405df657986"
}
```

The check has to rely on the mandatory fields being present (like `"paymentHash"`).

Given the API, there isn't anything else to be done.